src/lsstseries/analysis/structure_function/base_calculator.py

```diff
@@ -189,10 +189,12 @@ def _calculate_binned_statistics(self, sample_values=None, statistic_to_apply="m
                 if len(self._dts[lc_idx]) > 1:
                     # bin the delta_time values, and evaluate the `statistic_to_apply`
                     # for the delta_flux values in each bin.
-                    self._bin_dts(self._dts[lc_idx])
+
+                    if self._bins is None:
```

Comment:
This is the line that fixes the bug.

Reply:
This is the bug fix, and I see you removed some unused lines.
Do the three unit tests you added test this bug? It isn't obvious to me so I ask for confirmation.

Reply:
The last unit test in test_analysis is specifically protecting this change from regressions. The other tests are just for increased code coverage.

Reply:
I added a comment on the line in `test_analysis` that is actually doing the work :)